ldap/servers/slapd/daemon.c

```diff
@@ -775,6 +794,83 @@ disk_monitoring_thread(void *nothing __attribute__((unused)))
         return;
 }
 
+char *epoll_event_flags_to_string(PRUint32 events)
+{
+    static char buf[64];
+    int len = 0;
+
+    if (events & EPOLLIN) {
+        len += snprintf(buf + len, sizeof(buf) - len, "EPOLLIN|");
+    }
+    if (events & EPOLLOUT) {
+        len += snprintf(buf + len, sizeof(buf) - len, "EPOLLOUT|");
+    }
+    if (events & EPOLLERR) {
+        len += snprintf(buf + len, sizeof(buf) - len, "EPOLLERR|");
+    }
+    if (events & EPOLLHUP) {
+        len += snprintf(buf + len, sizeof(buf) - len, "EPOLLHUP|");
+    }
+    if (events & EPOLLRDHUP) {
+        len += snprintf(buf + len, sizeof(buf) - len, "EPOLLRDHUP|");
+    }
+    if (events & EPOLLEXCLUSIVE) {
+        len += snprintf(buf + len, sizeof(buf) - len, "EPOLLEXCLUSIVE|");
+    }
+    if (events & EPOLLPRI) {
+        len += snprintf(buf + len, sizeof(buf) - len, "EPOLLPRI|");
+    }
+    if (events & EPOLLONESHOT) {
+        len += snprintf(buf + len, sizeof(buf) - len, "EPOLLONESHOT|");
+    }
+    if (events & EPOLLET) {
+        len += snprintf(buf + len, sizeof(buf) - len, "EPOLLET|");
+    }
+    if (len == 0) {
+        len += snprintf(buf + len, sizeof(buf) - len, "EPOLLUNKNOWN|");
+    }
+    /* Remove the trailing | */
+    if (len > 0 && buf[len - 1] == '|') {
+        buf[len - 1] = '\0';
+    }
+    /* If no flags were set, return a specific message */
+    if (len == 0 || (len == 1 && buf[0] == '\0')) {
+        snprintf(buf, sizeof(buf), "No flags");
+    }
+    return buf;
+}
+
+#ifdef ENABLE_EPOLL
+static void
+handle_listeners(struct epoll_event *events, int event_count)
+{
+    Connection_Table *ct = the_connection_table;
+    size_t idx;
+    int ctlist = 0;
+    struct listener_info *listener = NULL;
+    for (idx = 0; idx < (size_t)event_count; idx++) {
```

Comment:
minor, `for (size_t idx = 0; ... )` There are some compiler optimisations that only work if you put it into the for loop like this, and it's a nice defence against using the idx outside the loop generally.